core/src/plugins/filed/python/proxmox/bareos-fd-proxmox.py

```diff
@@ -0,0 +1,339 @@
+            200,
+            (
+                "plugin_io() called with function %s"
+                " IOP.count=%s, self.FNAME is set to %s\n"
+            )
+            % (IOP.func, IOP.count, self.FNAME),
+        )
+
+        if IOP.func == bareosfd.IO_OPEN:
+            self.FNAME = IOP.fname
+            bareosfd.DebugMessage(
+                100, "IO_OPEN: %s\n" % (self.FNAME)
+            )
+            if self.options.get("restoretodisk") == "yes":
+                IOP.status = bareosfd.iostat_do_in_core
+                self.file = open(self.FNAME, "wb")
+                IOP.filedes = self.file.fileno()
+                bareosfd.JobMessage( bareosfd.M_INFO, f"restoring to file {self.FNAME}\n")
+                return bareosfd.bRC_OK
+            #return super(BareosFdProxmox, self).plugin_io(IOP)
+
+            # TODO: Check if restore or backup to determine
+            if self.vzdump_process.stdin:
+                IOP.filedes = self.vzdump_process.stdin.fileno()
+            else:
+                IOP.filedes = self.vzdump_process.stdout.fileno()
+            IOP.status = bareosfd.iostat_do_in_core
+
+            return bareosfd.bRC_OK
+
+        elif IOP.func == bareosfd.IO_CLOSE:
+            self.FNAME = IOP.fname
+            bareosfd.DebugMessage(
+                100, "IO_CLOSE: %s\n" % (self.FNAME)
+            )
+
+            if self.options.get("restoretodisk") == "yes":
+                self.file.close()
+                return bareosfd.bRC_OK
+
+            for line in self.current_logfile.readlines():
+                bareosfd.JobMessage( bareosfd.M_INFO, line)
+
+            # check for process return code
+            if self.vzdump_process.returncode:
+                bareosfd.DebugMessage(
+                100,
+                "plugin_io() bareos_vadp_dumper returncode: %s\n"
+                % (bareos_vadp_dumper_returncode),
+                )
+                return bareosfd.bRC_ERR
+
+            return bareosfd.bRC_OK
+
+        elif IOP.func == bareosfd.IO_SEEK:
+            bareosfd.DebugMessage(
+                100, "IO_SEEK: %s\n" % (self.FNAME)
+            )
+            return bareosfd.bRC_OK
+
+        elif IOP.func == bareosfd.IO_READ:
+            bareosfd.DebugMessage(
+                100, "IO_READ: %s\n" % (self.FNAME)
+            )
+            return bareosfd.bRC_OK
+
+        elif IOP.func == bareosfd.IO_WRITE:
+            bareosfd.DebugMessage(
+                100, "IO_WRITE: %s\n" % (self.FNAME)
+            )
+            try:
+                self.vzdump_process.stdin.write(IOP.buf)
+                IOP.status = IOP.count
+                IOP.io_errno = 0
+
+            except IOError as e:
+                bareosfd.DebugMessage(100, "plugin_io[IO_WRITE]: IOError: %s\n" % (e))
+                #self.vadp.end_dumper()
+                IOP.status = 0
+                IOP.io_errno = e.errno
+                return bareosfd.bRC_Error
+
+            return bareosfd.bRC_OK
```

Comment:
These should never be called when doing I/O in core, so you can just do a fatal job message for these cases of `IOP.func` and be done with it.